### PR TITLE
Remove statistics-based set_index logic from read_parquet

### PR DESCRIPTION
**Pull request by rjzamora**
`dd.read_parquet` will only gather statistics for columns that either (1) have been designated as an index or (2) are being filtered. For this reason, it no longer makes sense to auto-infer an index that was not specified in the pandas metadata or by the user (via `dd.read_parquet(..., index=)`). By using statistics to automatically set an arbitrary sorted column as the index, we open ourselves up to problematic/surprising behavior (see https://github.com/dask-contrib/dask-sql/pull/903#discussion_r1022035461).

This PR proposes that we officially remove the logic use to automatically select an index column using statistics.

- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by rjzamora** [comment]
cc @charlesbluca

**Review by charlesbluca — APPROVED** [review]

**Comment by fjetter** [comment]
+1 for the change itself. I don't think we should auto set the index. From a UX perspective, I could see this being a utility function answering the question "Are there sorted columns that could be useful to be an index in my dataset" but setting it automatically without opt-out appears to be strange.

This is likely breaking to some. I'm wondering if we want to communicate this somehow. How did we deal with these situations in the past?

Code also LGTM. I'm OK with merging after a brief discussion about deprecation cylce

**Comment by rjzamora** [comment]
>This is likely breaking to some. I'm wondering if we want to communicate this somehow. How did we deal with these situations in the past?

Good point. Although this PR changes the code to be **more** consistent with documentation, I agree that some users *may* be expecting the "undocumented" behavior that we are trying to fix.

The good news is that the only case we are really changing here is when `index=None` and `filters` is defined. This is because we only collect min/max statistics for index columns and filtered columns anyway. Therefore, I was able to add a simple `UserWarning` for the case that we have detected a sorted column that is **not** being used as the index already (and `index=None`).

File: dask/dataframe/io/tests/test_parquet.py [hunk 1]
```diff
@@ -4401,3 +4401,14 @@ def test_retries_on_remote_filesystem(tmpdir):
...
+
+def test_select_filtered_column(tmp_path, engine):
+
+    df = pd.DataFrame({"a": range(10), "b": ["cat"] * 10})
+    path = tmp_path / "test_select_filtered_column.parquet"
+    df.to_parquet(path, index=False)
+
+    with pytest.warns(UserWarning, match="Sorted columns detected"):
+        ddf = dd.read_parquet(path, engine=engine, filters=[("b", "==", "cat")])
+    assert_eq(df, ddf)
```
Comment (fjetter) [inline dask/dataframe/io/tests/test_parquet.py:4414]: The fact that you are deleting so much logic and you are not even deleting a test but add a new one is possibly the most compelling reason why this _feature_ needs to go

**Review by fjetter — APPROVED** [review]
I like the suggestion with the user warning

**Pull request merged**